hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -163,6 +169,111 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     }
   }
 
+  @Override
+  public boolean createNamespace(Namespace namespace) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Namespace could not empty. ");
+    Preconditions.checkArgument(namespace.levels().length == 1,
+        "Hive MetaStore cannot support multi part namespace now: %s", namespace.toString());
+    try {
+      clients.run(client -> {
+        client.createDatabase(nameSpaceToHiveDb(namespace));
+        return null;
+      });
+
+      return true;
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException("namespace already exists: %s",
+            namespace.toString());
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to createDatabase(name)", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    List<Namespace> namespaces = new ArrayList<>();
+    List<String> dbs;
+    try {
+      if (namespace.isEmpty()) {
+        dbs = clients.run(HiveMetaStoreClient::getAllDatabases);
+      } else {
+        dbs = clients.run(client -> client.getDatabases(namespace.toString()));
+      }
+      for (String db : dbs) {
+        namespaces.add(Namespace.of(db));
+      }
+      return namespaces;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all namespace: " + e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to getAllDatabases()", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces() {
+    return listNamespaces(Namespace.empty());
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
```

Comment:
Spark will check whether a namespace is empty and reject the operation unless the user requested `CASCADE`. If `dropNamespace` is called by Spark, then it is okay to drop the entire namespace and its contents. So that should be the behavior here.